src/components/organisms/BannerAdmin/BannerAdmin.tsx

```diff
@@ -1,90 +1,176 @@
-import React, { useState, useCallback, useRef } from "react";
+import React, { useCallback, useState, useEffect } from "react";
+import { useForm } from "react-hook-form";
 
 import { makeUpdateBanner } from "api/bannerAdmin";
 
 import { AnyVenue } from "types/venues";
+import { Banner } from "types/banner";
+
+import { useShowHide } from "hooks/useShowHide";
+
+import { Button } from "components/atoms/Button";
+import { Checkbox } from "components/atoms/Checkbox";
+import { InputField } from "components/atoms/InputField";
+
+import { ConfirmationBannerModal } from "./ConfirmationBannerModal";
+
+import "./BannerAdmin.scss";
 
 interface BannerAdminProps {
   venueId?: string;
   venue: AnyVenue;
+  onClose: () => void;
 }
 
 // @debt This component is almost exactly the same as IframeAdmin, we should refactor them both to use the same generic base component
 //   BannerAdmin is the 'canonical example' to follow when we do this
-export const BannerAdmin: React.FC<BannerAdminProps> = ({ venueId, venue }) => {
-  const existingBannerMessage = venue?.bannerMessage ?? "";
-
-  const inputFieldRef = useRef<HTMLInputElement>(null);
-  const [error, setError] = useState<string | null>();
+export const BannerAdmin: React.FC<BannerAdminProps> = ({
+  venueId,
+  venue,
+  onClose,
+}) => {
+  const { register, handleSubmit, errors, reset, watch } = useForm<Banner>({
+    mode: "onChange",
+    reValidateMode: "onChange",
+  });
+  const isActionButtonValue = watch("isActionButton");
+
+  const [bannerData, setBannerDate] = useState<Banner>();
+
+  const {
+    isShown: isShowModal,
+    show: showModal,
+    hide: hideModal,
+  } = useShowHide();
+  const [isDisabledUrlFields, setIsDisabledUrlFields] = useState(
+    venue?.banner?.isActionButton
+  );
 
-  const handleInputChange = useCallback(() => {
-    setError(null);
-  }, []);
+  useEffect(() => {
+    setIsDisabledUrlFields(isActionButtonValue);
+  }, [isActionButtonValue]);
 
   const updateBannerInFirestore = useCallback(
-    (msg?: string) => {
+    (banner?: Banner) => {
       if (!venueId) return;
-
-      makeUpdateBanner(venueId, (errorMsg) => setError(errorMsg))(msg);
+      makeUpdateBanner({ venueId, banner });
+      onClose();
     },
-    [venueId]
+    [venueId, onClose]
   );
 
-  const saveBanner = useCallback(
-    (e: React.MouseEvent<HTMLButtonElement>) => {
-      e.preventDefault();
-
-      if (!inputFieldRef.current) return;
-
-      updateBannerInFirestore(inputFieldRef.current.value);
+  const onSubmit = useCallback(
+    (data: Banner) => {
+      showModal();
+      setBannerDate(data);
     },
-    [updateBannerInFirestore]
+    [showModal]
   );
```

Comment:
This used to use a semantically relevant 'action name' (`saveBanner`), but now it's been renamed to use a passive name (`onSubmit`). Why is that?

Reply:
not sure, tho changed the naming back to `saveBanner`